Sources/AEPOptimize/Optimize.swift

```diff
@@ -139,6 +128,43 @@ public class Optimize: NSObject, Extension {
         return true
     }
 
+    /// Processes the propositions request event, dispatched with type `EventType.optimize` and source `EventSource.requestContent`.
+    ///
+    /// It processes events based on the type of the requested event type
+    /// - Parameter event: propositions request event
+    private func processOptimizeRequestContent(event: Event) {
+        if event.isUpdateEvent {
+            processUpdatePropositions(event: event)
+        } else if event.isGetEvent {
+            guard let decisionScopes: [DecisionScope] = event.getTypedData(for: OptimizeConstants.EventDataKeys.DECISION_SCOPES),
+                  !decisionScopes.isEmpty
+            else {
+                /// If decision scopes are not present in the event data, then adding it to the event queue
+                eventsQueue.add(event)
+                Log.warning(label: OptimizeConstants.LOG_TAG, "Decision scopes are either not present or empty in the event data.")
+                return
+            }
+            /// Fetch propositions and check if all of the decision scopes are present in the cache
+            let fetchedPropositions = decisionScopes.filter { self.cachedPropositions.keys.contains($0) }
+            /// Check if the decision scopes are currently in progress in `updateRequestEventIdsInProgress`
+            let scopesInProgress = decisionScopes.filter { scope in
+                updateRequestEventIdsInProgress.values.flatMap { $0 }.contains(scope)
+            }
+            if decisionScopes.count == fetchedPropositions.count && scopesInProgress.isEmpty {
+                processGetPropositions(event: event)
+            } else {
+                /// Not all decision scopes are present in the cache or requested scopes are currently in progress, adding it to the event queue
+                eventsQueue.add(event)
+                Log.warning(label: OptimizeConstants.LOG_TAG, "Decision scopes are either not present or currently in progress.")
```

Comment:
same here, a `trace` explaining to the developer that there are in-progress requests for the given scope and that the get request will be added to the queue accordingly would be appropriate